src/shared/ui/components/BaseTabs.test.ts

```diff
@@ -0,0 +1,62 @@
+import { expect, test } from 'vitest';
+import { ref } from 'vue';
+import { mount } from '@vue/test-utils';
+import type { BlockTabs } from '@/features/filter-transactions/model';
+import { blockOptions } from '@/features/filter-transactions/model';
+import { adaptiveTransactionTypeOptions } from '@/pages/Accounts/consts';
+import { i18n } from '@/shared/lib/localization';
+import BaseTabs from '@/shared/ui/components/BaseTabs.vue';
+
+test.each([
+  [1700, 6],
+  [1440, 5],
+  [1200, 3],
+  [960, 6],
+  [640, 3],
+  [480, 5],
+  [365, 2],
+])('BaseTabs adaptive display correctness', async (windowWidth, expectedTabs) => {
+  const model = ref<BlockTabs>('transactions');
+
+  window.innerWidth = windowWidth;
+
+  const wrapper = mount(BaseTabs, {
+    props: {
+      items: blockOptions,
+      modelValue: model.value,
+      adaptiveOptions: adaptiveTransactionTypeOptions,
+    },
+    global: {
+      plugins: [i18n],
+    },
+  });
+
+  function checkTabs() {
+    expect(wrapper.findAll('.base-tabs__tab').length).toBe(expectedTabs);
+  }
+
+  if (expectedTabs === 2) {
+    const nextButton = wrapper.find(`[data-testid="next"]`);
+
+    await nextButton.trigger('click');
+    checkTabs();
+
+    await nextButton.trigger('click');
+    checkTabs();
+
+    const prevButton = wrapper.find(`[data-testid="prev"]`);
+    await prevButton.trigger('click');
+
+    checkTabs();
+  } else if (expectedTabs !== 6) {
```

Comment:
And what if it is 6? Then the test will check nothing?

Reply:
According to layouts the max length of tabs is 6. Hence, when expected amount of tabs is 6 we don't display arrows. Otherwise we test arrows swiping correctness.

Reply:
Then it makes sense to ensure that arrows aren't displayed when tabs amount is 6. Otherwise it is confusing.